modules/cudf/src/series/integral.ts

```diff
@@ -190,6 +191,108 @@ abstract class IntSeries<T extends Integral> extends NumericSeries<T> {
   bitInvert(memoryResource?: MemoryResource): Series<T> {
     return Series.new(this._col.bitInvert(memoryResource));
   }
+
+  protected _prepare_scan_series(skipna: boolean) {
+    if (skipna || !this.hasNulls) { return this; }
+
+    const index = Series.sequence({type: new Int32, size: this.length, step: 1, init: 0});
+
+    const first = index.filter(this.isNull()).getValue(0)!;
+    const slice =
+      Series.sequence({type: new Int32, size: this.length - first, step: 1, init: first});
+
+    const copy = this.cast(this.type);
+    const mask = [...index.cast(new Bool8).fill(true).scatter(false, slice)];
+    copy.setNullMask(mask as any);
+
+    return copy;
+  }
+
+  /**
+   * Compute the cumulative max of all values in this Series.
+   *
+   * @param skipna The optional skipna if true drops NA and null values before computing reduction,
+   * else if skipna is false, reduction is computed directly.
+   * @param memoryResource The optional MemoryResource used to allocate the result Series's device
+   *   memory.
+   * @returns The cumulative max of all the values in this Series.
+   * @example
+   * ```typescript
+   * import {Series} from '@rapidsai/cudf';
+   * const a = Series.new([4, 2, 5, 1, 1])
+   *
+   * a.cummax() // {4, 4, 5, 5, 5}
```

Comment:
Should these values inside the array be big ints? i.e `4` -> `4n`?

Reply:
bigint output only for bigint input AFAIK. That said, `Series.new([4, 2, 5, 1, 1])` actually creates a Float32 series. It would be more precise but tedious to to`Series.new({type: new Int32, data: [4, 2, 5, 1, 1]})` which would yield the current output. 

Or, change to `a = Series.new([4n, 2n, 5n, 1n, 1n])` input to get the bigint output you suggest. 

@trxcllnt do you have a preference on which to demo in the docstrings? (or both?)

Reply:
Ah right, that makes sense. Thanks.

Reply:
per @trxcllnt just adding `.cast(new Int32)` is also nice